src/control/cmd/daos_server/start.go

```diff
@@ -0,0 +1,156 @@
+	SocketDir   string  `short:"d" long:"socket_dir" description:"Location for all daos_server & daos_io_server sockets"`
+	Insecure    bool    `short:"i" long:"insecure" description:"allow for insecure connections"`
+}
+
+func (cmd *startCmd) setCLIOverrides() error {
+	// Override certificate support if specified in cliOpts
+	if cmd.Insecure {
+		cmd.config.TransportConfig.AllowInsecure = true
+	}
+	if cmd.Port > 0 {
+		cmd.config.Port = int(cmd.Port)
+	}
+	if cmd.MountPath != "" {
+		cmd.config.ScmMountPath = cmd.MountPath
+	}
+	if cmd.Group != "" {
+		cmd.config.SystemName = cmd.Group
+	}
+	if cmd.SocketDir != "" {
+		cmd.config.SocketDir = cmd.SocketDir
+	}
+	if cmd.Modules != nil {
+		cmd.config.Modules = *cmd.Modules
+	}
+	if cmd.Attach != nil {
+		cmd.config.Attach = *cmd.Attach
+	}
+
+	host, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+	if cmd.config.NvmeShmID == 0 {
+		cmd.config.SetNvmeShmID(host)
+	}
+
+	for _, srv := range cmd.config.Servers {
+		srv.Hostname = host
+
+		if cmd.MountPath != "" {
+			// override each per-server config in addition to global value
+			srv.ScmMount = cmd.MountPath
+		} else if srv.ScmMount == "" {
+			// if scm not specified for server, apply global
+			srv.ScmMount = cmd.config.ScmMountPath
+		}
+		if cmd.Targets > 0 {
+			srv.Targets = int(cmd.Targets)
+		}
+		if cmd.NrXsHelpers != nil {
+			srv.NrXsHelpers = int(*cmd.NrXsHelpers)
+		}
+		if cmd.FirstCore > 0 {
+			srv.FirstCore = int(cmd.FirstCore)
+		}
+	}
+
+	return cmd.config.SetIOServerFlags()
+}
+
+func (cmd *startCmd) configureLogging() error {
+	// Set log level mask for default logger from config,
+	// unless it was explicitly set to debug via CLI flag.
+	applyLogConfig := func() {
+		switch logging.LogLevel(cmd.config.ControlLogMask) {
+		case logging.LogLevelDebug:
+			cmd.log.SetLevel(logging.LogLevelDebug)
+			cmd.log.Debugf("Switching control log level to DEBUG")
+		case logging.LogLevelError:
+			cmd.log.Debugf("Switching control log level to ERROR")
+			cmd.log.SetLevel(logging.LogLevelError)
+		}
+
+		if cmd.config.ControlLogJSON {
+			cmd.log = cmd.log.WithJSONOutput()
+		}
+	}
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+
+	// Set log file for default logger if specified in config.
+	if cmd.config.ControlLogFile != "" {
+		f, err := common.AppendFile(cmd.config.ControlLogFile)
+		if err != nil {
+			return errors.WithMessage(err, "create log file")
+		}
+
+		cmd.log.Infof("%s logging to file %s",
+			os.Args[0], cmd.config.ControlLogFile)
+		// Create an additional set of loggers which append everything
+		// to the specified file.
+		cmd.log = cmd.log.
+			WithErrorLogger(logging.NewErrorLogger(hostname, f)).
+			WithInfoLogger(logging.NewInfoLogger(hostname, f)).
+			WithDebugLogger(logging.NewDebugLogger(f))
+		applyLogConfig()
+
+		return nil
+	}
+
+	cmd.log.Info("no control log file specified; logging to stdout")
+	applyLogConfig()
+
+	for i, srv := range cmd.config.Servers {
```

Comment:
this check is not performed if ControlLogFile is specified, whereas it should be regardless of whether the control log file is specified

Reply:
This function was moved mostly-unchanged from server/config.go: https://github.com/daos-stack/daos/blob/master/src/control/server/config.go#L341

If this is an existing bug to be fixed, let's create a ticket for it. The ioserver configuration stuff is going to be changed in the near future anyhow (follow-on work to this).

Reply:
it's due to the change you applied, returning after checking for the control log file. I don't think there was a return before

Reply:
address in https://github.com/daos-stack/daos/pull/981